src/map/Map.js

```diff
@@ -180,11 +180,8 @@ export const Map = Evented.extend({
 		this._stop();
 
 		if (this._loaded && !options.reset && options !== true) {
-
-			if (options.animate !== undefined) {
-				options.zoom = Util.extend({animate: options.animate}, options.zoom);
-				options.pan = Util.extend({animate: options.animate, duration: options.duration}, options.pan);
-			}

```

Comment:
@mourner do you know, why if `if (options.animate !== undefined) {` exists? The problem is that the options like the duration are not applied until `animate: true` is passed in the options. `map.setView([50.5, 0], 9, {duration: 10, animate: true}); `

https://github.com/Leaflet/Leaflet/commit/260a869da6e6519a883bf733c7ca7caa34ce1540

Reply:
@Falke-Design the original change comes in #1617, and looking at the description, I think the logic was that without explicit `animate: true`, you have no guarantee that it will animate (it could instantly reset to new location if it's far away enough), so instead the behavior is fully heuristic instead of being based on specific options. But I'm not fully sure, this PR was more than 10 years ago.